core/src/main/scala/org/apache/spark/deploy/history/SimpleFsHistoryProvider.scala

```diff
@@ -0,0 +1,273 @@
+          ui.getSecurityManager.setAdminAcls(adminAcls)
+          ui.getSecurityManager.setViewAcls(attemptInfo.sparkUser, appListener.viewAcls.getOrElse(""))
+          val adminAclsGroups = HISTORY_UI_ADMIN_ACLS_GROUPS + "," +
+            appListener.adminAclsGroups.getOrElse("")
+          ui.getSecurityManager.setAdminAclsGroups(adminAclsGroups)
+          ui.getSecurityManager.setViewAclsGroups(appListener.viewAclsGroups.getOrElse(""))
+              
+          LoadedAppUI(ui, () => false)
+        })
+      })
+    } catch {
+        case e: Exception =>
+          logError(s"Exception encountered when attempting to load application log ${logDir}/${appId}", e)
+          None
+        case e: FileNotFoundException =>
+          logError(s"File Not Found when trying to load log ${logDir}/${appId}", e)
+          None
+    }
+  }
+  
+ 
+  /**
+   * Replays the events in the specified log file on the supplied `ReplayListenerBus`. Returns
+   * an `ApplicationEventListener` instance with event data captured from the replay.
+   * `ReplayEventsFilter` determines what events are replayed and can therefore limit the
+   * data captured in the returned `ApplicationEventListener` instance.
+   */
+  private def replay(
+      eventLog: FileStatus,
+      appCompleted: Boolean,
+      bus: ReplayListenerBus,
+      eventsFilter: ReplayEventsFilter = SELECT_ALL_FILTER): ApplicationEventListener = {
+    val logPath = eventLog.getPath()
+    logInfo(s"Replaying log path: $logPath")
+    // Note that the eventLog may have *increased* in size since when we grabbed the filestatus,
+    // and when we read the file here.  That is OK -- it may result in an unnecessary refresh
+    // when there is no update, but will not result in missing an update.  We *must* prevent
+    // an error the other way -- if we report a size bigger (ie later) than the file that is
+    // actually read, we may never refresh the app.  FileStatus is guaranteed to be static
+    // after it's created, so we get a file size that is no bigger than what is actually read.
+    val logInput = EventLoggingListener.openEventLog(logPath, fs)
+    try {
+      val appListener = new ApplicationEventListener
+      bus.addListener(appListener)
+      bus.replay(logInput, logPath.toString, !appCompleted, eventsFilter)
+      appListener
+    } finally {
+      logInput.close()
+    }
+  }
+ 
+  override def writeEventLogs(
+      appId: String,
+      attemptId: Option[String],
+      zipStream: ZipOutputStream): Unit = {
+
+    /**
+     * This method compresses the files passed in, and writes the compressed data out into the
+     * [[OutputStream]] passed in. Each file is written as a new [[ZipEntry]] with its name being
+     * the name of the file being compressed.
+     */
+    def zipFileToStream(file: Path, entryName: String, outputStream: ZipOutputStream): Unit = {
+      val fs = file.getFileSystem(hadoopConf)
```

Comment:
do we need to get another FS?

Reply:
This is part of the untouched code from FsHistoryProvider, but it looks like you make Path  object, then call get getFileSystem.  Seems kind of backwards, but that appears to be how it works.  This FS object will be specific to the application's event log's dir vs the whole event-logs logdir

Reply:
Makes sense now - still looks D: